src/charm.py

```diff
@@ -112,14 +131,37 @@ def _configure(self, _):
             reloaded = self._prometheus_server.reload_configuration()
             if not reloaded:
                 self.unit.status = BlockedStatus("Failed to load Prometheus config")
-            else:
-                self.unit.status = ActiveStatus()
-                logger.info("Prometheus configuration reloaded")
+                return
+            logger.info("Prometheus configuration reloaded")
         else:
             container.add_layer(self._name, new_layer, combine=True)
             container.restart(self._name)
             logger.info("Prometheus (re)started")
-            self.unit.status = ActiveStatus()
+
+        # Ensure the right address is set on the remote_write relations
+        self.remote_write_provider.update_endpoint()
```

Comment:
Can this be delegated to library code?

Reply:
Moved to https://github.com/canonical/prometheus-k8s-operator/issues/195